src/Core/Context.idr

```diff
@@ -2143,6 +2143,14 @@ getWorkingDir
               | Nothing => throw (InternalError "Can't get current directory")
          pure d
 
+export
+withCtxt : {auto cc : Ref Ctxt Defs} -> Core a -> Core a
```

Comment:
I would expect a `withX` function to take an `(x : X)` and locally override the value
of a global `X` with this `x` à la 

```idris
withVerbosity : Int -> Core a -> Core a
```

to escalate the verbosity level of debug message; or provide a sanitized version of
this `x` à la

```idris
withFile : FilePath -> (Handle -> IO a) -> IO a
```

that makes sure to close the handle in case an exception is raised.

Should this be called `saveX` or something similar instead?

Reply:
I would argue that it does sanitize the`x`.
```
{c : _}  ->
(op: {c : _} -> Core a) ->
Core a
```
Both happen to have the same type, and we don't force the `Core a` to depend on the Ref, but that's what I see it doing: providing a locally mutable resource to `op`.

Reply:
ok